### PR TITLE
Feature/sensitivity low alert

### PR DESCRIPTION
**Pull request by skvorekn**
Sensitivity at low alert rates is an evaluation metric used in the final round of the [Centers for Medicare & Medicaid Services AI Health Outcomes Challenge](https://innovation.cms.gov/innovation-models/artificial-intelligence-health-outcomes-challenge).

It is an important clinical accuracy metric because it gives insight into how resource reallocation will impact outcomes. For example, if we are predicting mortality of the Medicare population, a hospital may want to allocate more resources to people with the top, say, 1% of risk scores (predictions). This 1% is the alert rate. As a measure of accuracy, the hospital may want to know what percent of all deaths are captured by focusing on the top 1% (sensitivity).

The sensitivity metric is calculated by classifying the top 1% of predictions as the 'True' class (we predict they will die), and the remaining 99% are classified as 'False' (we predict they will not die). Sensitivity is measured using this alert rate (# predicted true/# actual true).

### CODE REVIEW

**Comment by skvorekn** [comment]
> [CLA assistant check](https://cla-assistant.io/alteryx/evalml?pullRequest=2001)
> Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/alteryx/evalml?pullRequest=2001) before we can accept your contribution.
> **1** out of **2** committers have signed the CLA.
> 
> ✅ skvorekn
> ❌ Kelsey Skvoretz
> 
> **Kelsey Skvoretz** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/alteryx/evalml?pullRequest=2001) it.

Please advise on how to resolve this. I only have one GitHub account, so not sure what is going on here! Thanks!

File: evalml/tests/objective_tests/test_sla.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        assert self.objective.decision_function(ypred_proba).tolist() == high_risk
+
+    @pytest.mark.parametrize("y_true, y_predicted, expected_score", [
+        (pd.Series([False, False, False]), pd.Series([True, True, False]), np.nan),
+        (pd.Series([True, True, True, True]), pd.Series([True, True, False, False]), 0.5)])
+    def test_score(self, y_true, y_predicted, expected_score):
+        sensitivity = SLA(0.1).objective_function(y_true, y_predicted)
+        assert (sensitivity is expected_score) or (sensitivity == expected_score)
+
+    def test_all_base_tests(self, fix_y_pred_na, fix_y_true, fix_y_pred_diff_len, fix_empty_array, fix_y_pred_multi):
```
Comment (chukarsten) [inline evalml/tests/objective_tests/test_sla.py:46]: I like this testing framework!

**Review by chukarsten — APPROVED** [review]
Solid.  I like this PR and the testing framework.  Thanks a lot for the contribution!

File: evalml/objectives/sensitivity_low_alert.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+        return ypred_proba.astype(float) >= prob_thresh
+
+    def objective_function(self, y_true, y_predicted, **kwargs):
+        """Calculate sensitivity across all predictions, using the top alert_rate percent of observations as the predicted positive class
+
+        Arguments:
+            y_true (ww.DataColumn, pd.Series): True labels
+            y_predicted (ww.DataColumn, pd.Series): Predicted labels based on alert_rate
+            X
```
[the remark this answers is not in the document]
Comment (chukarsten) [reply evalml/objectives/sensitivity_low_alert.py:52]: @skvorekn I forgot to mention that this errant X should be addressed in changes.

[the remark this answers is not in the document]
Comment (skvorekn) [reply evalml/objectives/sensitivity_low_alert.py:52]: Ah, yes! Thank you!

**Comment by skvorekn** [comment]
I misunderstood the comment about squashing changes, so did that out of order. Looks like things are passing now, though! Also made a few minor updates based on @chukarsten's comments. Thanks for your feedback - I really appreciate it! Let me know if there is anything else.

File: evalml/objectives/sensitivity_low_alert.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+import numpy as np
+
+from .binary_classification_objective import BinaryClassificationObjective
+
+from evalml.utils.logger import get_logger
+
+logger = get_logger(__file__)
+
+
+class SLA(BinaryClassificationObjective):
```
Comment (dsherry) [inline evalml/objectives/sensitivity_low_alert.py:10]: @skvorekn could we name this something like `SensitivityLowAlert`, just to give users a bit more context at first glance?

**Review by dsherry — APPROVED** [review]
This is wonderful! The code looks great. My only comment was about the class name.

File: evalml/objectives/sensitivity_low_alert.py (hunk 3 above)
Comment (skvorekn) [reply evalml/objectives/sensitivity_low_alert.py]: Good call! Done!

**Comment by chukarsten** [comment]
@skvorekn I think you might need to resign our CLA to pass our CI check.  You were this close 🤏 to making it into the last release .

**Comment by skvorekn** [comment]
Done! Looks like the CLA needs a resign by @chukarsten as well!

**Comment by chukarsten** [comment]
@skvorekn merged!  We'll release it next week with 0.22.0.

**Pull request merged**